frontend/src/app/choreography/choreography.component.ts

```diff
@@ -310,15 +313,28 @@ export class ChoreographyComponent {
   }
 
   saveChoreography(): void {
-    localStorage.setItem('choreography', JSON.stringify(this.choreography));
+    const dialogRef = this.dialog.open(SaveChoreographyDialogComponent, {});
+    dialogRef.afterClosed().subscribe(result => {
+      if (result) {
```

Comment:
**suggestion:** I think a slightly cleaner way to do this would be `dialogRef.afterClosed().pipe(filter(result => !!result).subscribe(...)`